org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/staging/AbapGitStagingView.java

```diff
@@ -702,8 +707,17 @@ private IStatus fetchCredentialsAndRefresh() {
 		if (this.repositoryExternalInfo != null) {
 			//get credentials from user if the repository is private
 			if (checkIfCredentialsRequired()) {
-				if (getGitCredentials().equals(Status.CANCEL_STATUS)) {
-					return Status.CANCEL_STATUS;
+				this.repositoryCredentials = getRepoCredentialsFromSecureStorage(this.repository.getUrl());
```

Comment:
No credentials dialog will be involved and the credentials retrieved from SecureStorage will be directly used to perform refresh action.